src/components/ManaCalculator/hooks/useManaState.ts

```diff
@@ -162,7 +171,27 @@ export function useGivenManaState(
   }
 
   function handleOwnHoldChange(value: number) {
-    setState({ ...state, heldTokens: value });
+    setState({
+      ...state,
+      ...getDerivedValuesFromHeldTokens(value),
+    });
+  }
+
+  function getDerivedValuesFromHeldTokens(
+    heldTokens: number,
+  ): Partial<ManaCalculatorProps> {
+    return {
+      heldTokens,
+      [getStakedOrDelegated(state.userType)]: heldTokens,
+      ...(state.userType === UserType.VALIDATOR
+        ? {
+            validator: {
+              ...state.validator,
+              attractedNewDelegatedStake: heldTokens,
+            },
+          }
+        : {}),
+    };
```

Comment:
this whole section is 99% duplicated to what is in lines 157-164, can we avoid code duplication somehow? Or is this needed?